connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/DockerJDBCIntegrationSuite.scala

```diff
@@ -129,104 +130,129 @@ abstract class DockerJDBCIntegrationSuite
+                s"Timeout('$imagePullTimeout secs') waiting for image ${db.imageName} to be pulled")
+            }
+        }
 
-      val hostConfig = HostConfig
-        .newHostConfig()
-        .withNetworkMode("bridge")
-        .withPrivileged(db.privileged)
-        .withPortBindings(PortBinding.parse(s"$externalPort:${db.jdbcPort}"))
+        val hostConfig = HostConfig
+          .newHostConfig()
+          .withNetworkMode("bridge")
+          .withPrivileged(db.privileged)
+          .withPortBindings(PortBinding.parse(s"$externalPort:${db.jdbcPort}"))
 
-      if (db.usesIpc) {
-        hostConfig.withIpcMode("host")
-      }
+        if (db.usesIpc) {
+          hostConfig.withIpcMode("host")
+        }
 
-      val containerConfig = new ContainerConfig()
+        val containerConfig = new ContainerConfig()
 
-      db.beforeContainerStart(hostConfig, containerConfig)
+        db.beforeContainerStart(hostConfig, containerConfig)
 
-      // Create the database container:
-      val createContainerCmd = docker.createContainerCmd(db.imageName)
-        .withHostConfig(hostConfig)
-        .withExposedPorts(ExposedPort.tcp(db.jdbcPort))
-        .withEnv(db.env.map { case (k, v) => s"$k=$v" }.toList.asJava)
-        .withNetworkDisabled(false)
+        // Create the database container:
+        val createContainerCmd = docker.createContainerCmd(db.imageName)
+          .withHostConfig(hostConfig)
+          .withExposedPorts(ExposedPort.tcp(db.jdbcPort))
+          .withEnv(db.env.map { case (k, v) => s"$k=$v" }.toList.asJava)
+          .withNetworkDisabled(false)
 
 
-      db.getEntryPoint.foreach(ep => createContainerCmd.withEntrypoint(ep))
-      db.getStartupProcessName.foreach(n => createContainerCmd.withCmd(n))
+        db.getEntryPoint.foreach(ep => createContainerCmd.withEntrypoint(ep))
+        db.getStartupProcessName.foreach(n => createContainerCmd.withCmd(n))
 
-      container = createContainerCmd.exec()
-      // Start the container and wait until the database can accept JDBC connections:
-      docker.startContainerCmd(container.getId).exec()
-      eventually(timeout(startContainerTimeout.seconds), interval(1.second)) {
-        val response = docker.inspectContainerCmd(container.getId).exec()
-        assert(response.getState.getRunning)
-      }
-      jdbcUrl = db.getJdbcUrl(dockerIp, externalPort)
-      var conn: Connection = null
-      eventually(connectionTimeout, interval(1.second)) {
-        conn = getConnection()
-      }
-      // Run any setup queries:
-      try {
-        dataPreparation(conn)
-      } finally {
-        conn.close()
-      }
-    } catch {
-      case NonFatal(e) =>
-        logError(s"Failed to initialize Docker container for ${this.getClass.getName}", e)
+        container = createContainerCmd.exec()
+        // Start the container and wait until the database can accept JDBC connections:
+        docker.startContainerCmd(container.getId).exec()
```

Comment:
Due to limited GHA resources, it is not advisable to repeatedly start containers.